framework/include/preference/preference.h

```diff
@@ -27,10 +27,24 @@
 #ifndef __FRAMEWORK_INCLUDE_PREFERENCE_PREFERENCE_H__
 #define __FRAMEWORK_INCLUDE_PREFERENCE_PREFERENCE_H__
 
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
 #include <stdbool.h>
 #include <tinyara/preference.h>
 
-#ifdef __cplusplus
+/****************************************************************************
+ * Pre-processor Definitions
```

Comment:
Enum does not come under preprocessor defines. 
Please change it to `Public Types `